keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +163,222 @@ Running this set of tests is optional for a per-provider basis. We would
+  - If feature parity is not guaranteed or if any bug exists in the CSI driver/csi-translation-lib, the rollout could fail because pod using the PV could fail to execute provision/delete/attach/detach/mount/unmount/resize operations depend on the bug itself.
+
+* **What specific metrics should inform a rollback?**
+  We have metrics on the CSI sidecar side called `csi_operation_duration_seconds` and core k8s metrics on both kube-controller-manager and kubelet side called `storage_operation_duration_seconds`. 
+  Both of them will have a `migrated` field to indicate whether indicate this operation is a migrated PV operation. 
+    - For `csi_operation_duration_seconds`, we will have a `grpc_status` field
+    - For `storage_operation_duration_seconds`, we will have a `status` field
+  
+  If the error ratio of these two metrics has an unusual strike or is keeping at a relatively higher level compared to in-tree model, it means something went wrong and we need a rollback.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+To turn it on by default in Beta, we require each in-tree plugin to at least manually test the upgrade->downgrade->upgrade path.
+For GA, we require such test exists in each driver's test CI.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+There will not be API removal in CSI migration itself. But eventually when CSI migration is all finished. We will plan to remove all in-tree plugins.
+So we will have in-tree plugin deprecated when CSIMigration{cloud-provider} goes to beta. And code removal will be required eventually.
+In addition, some CSI drivers are not able to maintain 100% backwards compatibility, so those drivers need to deprecate certain behaviors. For example, azurefile and vSphere has some corner cases that will not work anymore. (TODO: add more details.)
+
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  We will have metrics `csi_sidecar_duration_seconds` on the CSI sidecars and `storage_operation_duration_seconds` on the kube-controller-manager and kubelet side to indicate whether this operation is a migrated operation or not. These metrics will have a `migrated` field to indicate if this is a migrated operation.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [x] Metrics
+    - Metric name: csi_sidecar_duration_seconds && storage_operation_duration_seconds, these metrics will have a `migrated` field
+    - [Optional] Aggregation method:
+    - Components exposing the metric: CSI sidecars, kubelet, kube-controller-manager
+  - [x] Other (treat as last resort)
+    - Details: Pod using in-tree plugin has failure.
```

Comment:
How can you say if the pod is using in-tree plugin?

Reply:
If the pod is using a PVC that uses intree plugin. It should be able to tell from the PVC spec